test/cluster/cluster_redirection_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"github.com/hyperledger-labs/orion-server/internal/worldstate"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/hyperledger-labs/orion-server/pkg/types"
+	"github.com/hyperledger-labs/orion-server/test/setup"
+	"github.com/stretchr/testify/require"
+)
+
+// Scenario:
+// - Start a 3 node cluster with redirection allowed
+// - Submit txs to all nodes
+// - Tx redirection from follower to leader
+// - Stop nodes until there is no quorum => tx submission rejection
+func TestRedirectionWithNoQuorum(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(1)
+	setupConfig := &setup.Config{
+		NumberOfServers:     3,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return nil
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+	// wait for leader
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0, 1, 2)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	for i := 0; i < 3; i++ {
+		txID, rcpt, err := c.Servers[i].WriteDataTx(t, worldstate.DefaultDBName, "key-"+strconv.Itoa(i), []byte("data-"+strconv.Itoa(i)))
+		require.NoError(t, err)
+		require.NotNil(t, rcpt)
+		require.True(t, txID != "")
+		require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+		require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+		t.Logf("tx submitted: %s, %+v", txID, rcpt)
+	}
+
+	for i := 0; i < 3; i++ {
+		require.Eventually(t, func() bool {
+			dataEnv, err := c.Servers[i].QueryData(t, worldstate.DefaultDBName, "key-"+strconv.Itoa(i), "admin")
+			if dataEnv != nil && dataEnv.GetResponse().GetValue() != nil {
+				dataVal := dataEnv.GetResponse().GetValue()
+				require.Equal(t, dataVal, []byte("data-"+strconv.Itoa(i)))
+				t.Logf("data: %+v", dataVal)
+			}
+			return dataEnv != nil && dataEnv.GetResponse().GetValue() != nil && err == nil
+		}, 30*time.Second, 100*time.Millisecond)
+	}
```

Comment:
here you should check that all servers have all 3 values

Reply:
done